lib/src/repository/check_repository.dart

```diff
@@ -143,9 +143,16 @@ Future<VerifiedRepository?> checkRepository({
               '`$path` from the repository has no `repository` or `homepage` URL.');
         }
         final gitRepoUrl = Repository.tryParseUrl(gitRepoOrHomepage);
-        if (gitRepoUrl?.cloneUrl != parsedSourceUrl.cloneUrl) {
+        if (gitRepoUrl == null) {
+          return _PubspecMatch(
+              path,
+              true,
+              '`$path` from the repository has a `repository` or `homepage` field '
+              '"`$gitRepoOrHomepage`" but cannot be parsed as a repository URL.');
```

Comment:
We should be explicit about "how" it cannot be parsed. Otherwise I would find this message confusing